### PR TITLE
openai:compatible with other llm usage meta data

### PR DESCRIPTION
**Pull request by omgbbq**
- [ ] **PR message**:
    - **Description:** Compatible with other llm (eg: deepseek-chat, glm-4) usage meta data
    - **Issue:** N/A
    - **Dependencies:** no new dependencies added

- [ ] **Add tests and docs**: 
libs/partners/openai/tests/unit_tests/chat_models/test_base.py
```shell
cd libs/partners/openai
poetry run pytest tests/unit_tests/chat_models/test_base.py::test_openai_astream
poetry run pytest tests/unit_tests/chat_models/test_base.py::test_openai_stream
poetry run pytest tests/unit_tests/chat_models/test_base.py::test_deepseek_astream
poetry run pytest tests/unit_tests/chat_models/test_base.py::test_deepseek_stream
poetry run pytest tests/unit_tests/chat_models/test_base.py::test_glm4_astream
poetry run pytest tests/unit_tests/chat_models/test_base.py::test_glm4_stream
```

### CODE REVIEW

**Review by efriis — COMMENTED** [review]
could you add a test case where both content and usage_metadata are returned? I'm pretty sure this has the exact same bug as #23704

I think this if/else block would need more substantial refactoring to work with this.

Will mark as draft for now - feel free to mark ready again when that's fixed!

File: libs/partners/openai/langchain_openai/chat_models/base.py [hunk 1]
```diff
@@ -280,6 +280,54 @@ def _convert_delta_to_message_chunk(
...
+        else None
+    )
+
+    if len(choices) == 0:
+        # logprobs is implicitly None
+        generation_chunk = ChatGenerationChunk(
+            message=default_chunk_class(content="", usage_metadata=usage_metadata)
+        )
+        return generation_chunk
+
```
Comment (efriis) [inline libs/partners/openai/langchain_openai/chat_models/base.py:304]: the concerning case in the previous code was when `usage_metadata` AND `content` came back from the model at the same time

**Review by efriis — COMMENTED** [review]
Could you fix the lint errors in the new tests?

File: libs/partners/openai/langchain_openai/chat_models/base.py (hunk 1 above)
Comment (omgbbq) [reply libs/partners/openai/langchain_openai/chat_models/base.py:308]: Yes，I changed the code and fixed the lint errors.

**Pull request merged**